src/mscorlib/src/System/Diagnostics/Eventing/EventSource.cs

```diff
@@ -250,6 +254,10 @@ namespace System.Diagnostics.Tracing
     /// </remarks>
     public partial class EventSource : IDisposable
     {
+#if FEATURE_CORECLR && FEATURE_PAL   
+        private static EventLogger persistent_Listener = new EventLogger();
```

Comment:
Isn't this approach going to mean that this listener is always-on, always serializing all events for all sources out to native (which only then may determine that tracing isn't needed), etc.?  That seems prohibitively expensive.